e2e-tests/insurance/cypress/e2e/journeys/account/sign-in/enter-code/account-sign-in-create-code-dashboard-or-application-redirect.spec.js

```diff
@@ -0,0 +1,63 @@
+import header from '../../../../../../../partials/header';
+
+import { INSURANCE_ROUTES as ROUTES } from '../../../../../../../constants/routes/insurance';
+import dashboardPage from '../../../../../../../pages/insurance/dashboard';
+
+const {
+  START,
+} = ROUTES;
+
+context('Insurance - Account - Sign in - I want to go straight to my application if there is exactly one application in progress or the dashboard otherwise', () => {
+  before(() => {
+    cy.navigateToUrl(START);
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  describe('when there are no applications', () => {
+    it('should redirect to the dashboard', () => {
+      cy.completeSignInAndGoToDashboard({});
+      // sign out for next test
+      header.navigation.signOut().click();
+    });
+  });
+
+  describe('when there is one application', () => {
+    it('should redirect to the application directly', () => {
+      // creates new application and account
+      cy.completeSignInAndGoToApplication().then(({ referenceNumber }) => {
+        // sign out for next test
+        header.navigation.signOut().click();
+
+        // sign in again and complete OTP (without creating account)
+        cy.completeSignInAndOTP({ referenceNumber, shouldRedirectToApplication: true });
+
+        // sign out for next test
+        header.navigation.signOut().click();
```

Comment:
Do we need to be doing this? Seems it's

- signs in, goes to an application
- signs out
- signs in
- signs out

Reply:
yes we do.  Once we have the application, we need to sign out to test the redirect.  And then we need to sign out for it to do the next test or it is on the wrong page for `completeSignInAndGoToApplication` which will create another account/application but with 2 this time

Reply:
Can we add some more comments for this, worry it would be confusing for anyone picking up in the future. 

Maybe better to move the `header.navigation.signOut().click();` into the next decsribe block in a beforeEach?

Reply:
Can do this in the next PR in the interests of reducing the PR backlog